cxx-squid/src/main/java/org/sonar/cxx/preprocessor/ExpressionEvaluator.java

```diff
@@ -144,7 +145,8 @@ private BigInteger evalComplexAst(AstNode exprAst) {
     } else if ("functionlikeMacro".equals(nodeType)) {
       return evalFunctionlikeMacro(exprAst);
     } else {
-      throw new EvaluationException("Unknown expression type '" + nodeType + "'");
+      LOG.error("'evalComplexAst' Unknown expression type '" + nodeType + "' for AstExt '" + exprAst.getToken() + "', assuming 0");
+      return BigInteger.ZERO;
```

Comment:
some examples:

[00:41:24] :		 [Step 3/3] 00:41:24.284 ERROR - 'evalComplexAst' Unknown expression type 'expression' for AstExt 'IDENTIFIER: BOOST_PP_TUPLE_ELEM', assuming 0